assets/common/src/batch_deploy.py

```diff
@@ -0,0 +1,338 @@
+    try:
+        logger.info(f"Creating endpoint {endpoint_name}")
+        ml_client.begin_create_or_update(endpoint).wait()
+    except Exception as e:
+        raise AzureMLException._with_error(
+            AzureMLError.create(EndpointCreationError, exception=e)
+        )
+
+    try:
+        logger.info(f"Creating deployment {deployment}")
+        ml_client.batch_deployments.begin_create_or_update(deployment).wait()
+    except Exception as e:
+        raise AzureMLException._with_error(
+            AzureMLError.create(DeploymentCreationError, exception=e)
+        )
+
+    logger.info("Deployment successful.")
+
+    # set the deployment as default
+    try:
+        logger.info(f"Updating endpoint to make {deployment_name} as default deployment")
+        endpoint = ml_client.batch_endpoints.get(endpoint_name)
+        endpoint.defaults.deployment_name = deployment_name
+        ml_client.begin_create_or_update(endpoint).wait()
+
+        endpoint = ml_client.batch_endpoints.get(endpoint_name)
+    except Exception as e:
+        error_msg = f"Error occured while updating deployment - {e}"
+        raise Exception(error_msg)
+
+    logger.info(f"The default deployment is {endpoint.defaults.deployment_name}")
+    return endpoint, deployment
+
+
+@swallow_all_exceptions(logger)
+def main():
+    """Run main function."""
+    args = parse_args()
+    ml_client = get_mlclient()
+    # get registered model id
+    if args.registration_details:
+        model_info = {}
+        with open(args.registration_details) as f:
+            model_info = json.load(f)
+        model_id = model_info["id"]
+        model_name = model_info["name"]
+    elif args.model_id:
+        model_id = str(args.model_id)
+        model_name = model_id.split("/")[-3]
+    else:
+        raise Exception("Arguments model_id and registration_details both are missing.")
+
+    # Endpoint has following restrictions:
+    # 1. Name must begin with lowercase letter
+    # 2. Followed by lowercase letters, hyphen or numbers
+    # 3. End with a lowercase letter or number
+
+    # 1. Replace underscores and slashes by hyphens and convert them to lower case.
+    # 2. Take 21 chars from model name and append '-' & timstamp(10chars) to it
+
+    endpoint_name = re.sub("[^A-Za-z0-9]", "-", model_name).lower()[:21]
+    endpoint_name = f"{endpoint_name}-{int(time.time())}"
+    endpoint_name = endpoint_name
+
+    endpoint_name = args.endpoint_name if args.endpoint_name else endpoint_name
+    deployment_name = args.deployment_name if args.deployment_name else "default"
+
+    endpoint, deployment = create_endpoint_and_deployment(
+        ml_client=ml_client,
+        endpoint_name=endpoint_name,
+        deployment_name=deployment_name,
+        model_id=model_id,
+        args=args
+    )
+
+    if args.inference_payload:
+        print("Invoking inference with test payload ...")
+        try:
+            input = Input(type="uri_folder", path=rf"{args.inference_payload}")
```

Comment:
this can be URI_FILE as well as URI_FOLDER. Please add something like this: 

```
    if batch_test_file_path and os.path.exists(batch_test_file_path):
        test_data = Input(path=batch_test_file_path, type=AssetTypes.URI_FILE)
        submit_job(mlclient, ep, test_data)
    elif batch_test_folder_path and os.path.exists(batch_test_folder_path):
        test_data = Input(path=batch_test_file_path, type=AssetTypes.URI_FILE)
        submit_job(mlclient, ep, test_data)
    else:
        logger.warning("test data not provided. Skipping job submission")
```

Reply:
Added inference for both URI_FILE as well as URI_FOLDER